aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -113,11 +113,17 @@ private Completable drainMutationOutbox() {
             if (next == null) {
                 return Completable.complete();
             }
-            boolean itemFailedToProcess = !processOutboxItem(next)
-                .blockingAwait(ITEM_PROCESSING_TIMEOUT_MS, TimeUnit.MILLISECONDS);
-            if (itemFailedToProcess) {
+            try {
+                boolean itemFailedToProcess = !processOutboxItem(next)
+                    .blockingAwait(ITEM_PROCESSING_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+                if (itemFailedToProcess) {
+                    return Completable.error(new DataStoreException(
+                        "Timeout processing " + next, "Check your internet connection."
+                    ));
+                }
+            } catch (RuntimeException error) {
```

Comment:
should we additionally catch the base exception? `catch(Exception e)`

Reply:
From the blockingAwait docs, mentioned earlier, the only other checked exception that would pop out is a NullPointerException which may be diligent to add, although I've never witnessed it.

Otherwise all unchecked exceptions should get picked up by https://github.com/aws-amplify/amplify-android/pull/1743/files#diff-ca60e5fcab67ffd6928cc6cc642368d11198d4567f37905a1d2505b6b31547a8R104